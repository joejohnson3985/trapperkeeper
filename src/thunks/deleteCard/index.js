import { removeCard } from '../../actions';
import fetchData from '../fetchData';

export const deleteCard = (id) => {
  return async (dispatch) =>  {
    try {
      const url = `http://localhost:3000/api/v1/cards/${id}`;
      const options = {
        method: 'DELETE'
      }
      await fetchData(url, options);
      dispatch(removeCard(id));
    } catch (error) {

    }
  }
}

export default deleteCard;