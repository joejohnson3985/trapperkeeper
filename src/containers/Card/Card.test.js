import React from 'react';
import { Card, mapDispatchToProps } from './index';
import { shallow } from 'enzyme';

import deleteCard from '../../thunks/deleteCard';
jest.mock('../../thunks/deleteCard')

describe('Card', () => {
  describe('component', () => {
    let wrapper;

    let mockCard = {id:1, name:'Chores', list:[1,2,3]}

    beforeEach(() => {
      wrapper = shallow(
        <Card card={mockCard} 
              key={mockCard.id}/>
      )
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  })

  describe('mapDispatchToProps', () => {
    it('should call deleteCard with the correct params', () => {
      let mockId = 12323123
      let mockDispatch = jest.fn();
      let actionToDispatch = deleteCard(mockId)
      let mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.deleteCard(actionToDispatch)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
  
})
