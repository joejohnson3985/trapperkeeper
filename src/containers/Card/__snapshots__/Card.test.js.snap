// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match the snapshot 1`] = `
<div
  className="card-outline"
>
  <Link
    className="card"
    to="/notes/1"
  >
    <h1>
      Chores
    </h1>
    <p>
      items: 
      3
    </p>
  </Link>
  <img
    alt="Delete card"
    className="delete-btn"
    onClick={[Function]}
    src="delete_outline.svg"
  />
</div>
`;
